sae_lens/training/sparse_autoencoder.py

```diff
@@ -102,7 +102,10 @@ def __init__(
 
         self.setup()  # Required for `HookedRootModule`s
 
-    def forward(self, x: torch.Tensor, dead_neuron_mask: torch.Tensor | None = None):
+    def encode(
+        self, x: torch.Tensor, return_hidden_pre: bool = False
```

Comment:
nit: I feel like the style of having different return types based on an input param should be avoided, as it breaks typing and makes it hard to know what methods are returning. Could we just return 1 thing from this method? It seems unnecessary to return the hidden_pre anyway IMO since this is already hooked, so it's easy to access that value as needed using the hook

Reply:
I was mainly motivated by wanting to be able to re-use this functionality in `forward`. happy to modify this if you think there's a better way